From 0d72b6fb7d4aa5e55294eb3222e7156d3d75a9e7 Mon Sep 17 00:00:00 2001
From: Daniel P. Berrange <berrange@redhat.com>
Date: Mon, 17 Aug 2009 08:52:30 +0100
Subject: [PATCH] Disable sound cards when running sVirt

Temporary hack till PulseAudio autostart problems are sorted out when
SELinux enforcing (bz 486112)

Fedora-patch: libvirt-0.6.2-svirt-sound.patch
---
 src/qemu_conf.c |   17 ++++++++++++++++-
 1 files changed, 16 insertions(+), 1 deletions(-)

diff --git a/src/qemu_conf.c b/src/qemu_conf.c
index 1194e36..f42aeaa 100644
--- a/src/qemu_conf.c
+++ b/src/qemu_conf.c
@@ -795,6 +795,20 @@ int qemudBuildCommandLine(virConnectPtr conn,
     char domid[50];
     char *pidfile;
     const char *cpu = NULL;
+    int skipSound = 0;
+
+    if (driver->securityDriver &&
+        driver->securityDriver->name &&
+        STREQ(driver->securityDriver->name, "selinux") &&
+        getuid() == 0) {
+        static int soundWarned = 0;
+        skipSound = 1;
+        if (vm->def->nsounds &&
+            !soundWarned) {
+            soundWarned = 1;
+            VIR_WARN0("Sound cards for VMs are disabled while SELinux security model is active");
+        }
+    }
 
     uname_normalize(&ut);
 
@@ -1441,7 +1455,8 @@ int qemudBuildCommandLine(virConnectPtr conn,
     }
 
     /* Add sound hardware */
-    if (vm->def->nsounds) {
+    if (vm->def->nsounds &&
+        !skipSound) {
         int size = 100;
         char *modstr;
         if (VIR_ALLOC_N(modstr, size+1) < 0)
-- 
1.6.2.5

